### PR TITLE
feat: add k8s module

### LINKED ISSUE #123: Kubernetes package
I think it would be interesting to have a k8s package to fetch objects from the API.

### PR DESCRIPTION
**Pull request by luisdavim**
closes #123

### CODE REVIEW

**Comment by myzie** [comment]
In general this is looking good. You have the k8s package as its own Go module and have it on a build tag, which is what I would do.

**Comment by luisdavim** [comment]
@myzie I did some testing and made some updates, I think this PR is ready to be reviewed/merged.
There's more functionality that we can add later but this should cover the base CRUD use cases and we can wait for user feedback to add more stuff.

File: modules/kubernetes/k8s.go
Comment (myzie) [inline modules/kubernetes/k8s.go:367]:
```suggestion
	return object.NewBuiltinsModule("k8s", map[string]object.Object{
```

**Comment by myzie** [comment]
I confirmed a basic `k8s.get("pod.v1")` works for me with Docker for Mac's k8s setup. I don't have another cluster at the moment to test against, so I'll defer to you on confirming anything beyond that for now!

Were the `toolchain go1.21.0` go.mod entries getting in the way for you?

**Comment by luisdavim** [comment]
Yes, I tested against both real clusters and Kind clusters. 

The `toolchain` directives were preventing `gopls` from working for me.

**Review by myzie — APPROVED** [review]
Awesome 😄

**Pull request merged**
